src/ConnectedFieldArray.js

```diff
@@ -23,7 +24,7 @@ const createConnectedFieldArray = ({
 
   class ConnectedFieldArray extends Component {
     shouldComponentUpdate(nextProps) {
-      return size(this.props.value) !== size(nextProps.value)
+      return shallowCompare(this, nextProps)
```

Comment:
I'm pretty sure. I think this will fix #931.

@leesiongchan, can you write a test where there is a `_error` validation error for the array based on some quality of the values inside the array (e.g. if none of them have values), and verify that the array component rerenders when the error appears or disappears?